erpnext/controllers/stock_controller.py

```diff
@@ -21,7 +21,7 @@ def make_gl_entries(self, gl_entries=None, repost_future_gle=True, from_repost=F
 		if self.docstatus == 2:
 			delete_gl_entries(voucher_type=self.doctype, voucher_no=self.name)
 
-		if cint(frappe.defaults.get_global_default("auto_accounting_for_stock")):
+		if cint(frappe.db.get_value('Company', self.company, 'enable_perpetual_inventory')):
```

Comment:
this function is called so many times!

It should be `erpnext.is_perpetual_inventory_enabled(company)` and the result must be cached locally for the company.